components/molecules/Pagination/pagination.tsx

```diff
@@ -43,42 +45,63 @@ const Pagination = ({
     onPageChange(page - 1);
   };
 
+  const middlePages = Array.from({ length: Math.min(pages.length, pageSize) }, (_, index) =>
+    selected <= Math.floor(pageSize / 2)
```

Comment:
IMHO it is very hard to read this function, I tend to believe it must be refactored.

Reply:
I'll go ahead and put a comment in the code about it. We're probably refactoring that whole paging mechanism. This feature is the first feature to use the paging in that mode.